docs/docs/creating-dynamically-rendered-navigation.md

```diff
@@ -2,7 +2,175 @@
+      {name:'home',link:'/'},
+      {name:'page2',link:'/page-2'}
+    ]
+  },
+  plugins: []
+}
+```
+### Viewing the `siteMetadata` in GraphQL
+
+GraphQL can be used to query for information contained in the `siteMetadata` object located in your project's `gatsby-config.js`. In order to test this out you can start the `default-starter-project` in development mode by running `npm run develop`.
+
+Navigate to `http://localhost:8000/___graphql` in your browser to view the GraphiQL editor, which enables you to test GraphQL queries on the underlying APIs. Using the documentation explorer you can view the current GraphQL schema for your project which is an invaluable resource during development.
+
+Examining the available types in GraphQL you will notice that you can query `site`. This GraphQL type further returns the `siteMetadata` which needs to be accessed to create the dynamic navigation. At this point it is useful if you know a little GraphQL in order to extract the menu links. If you are unfamiliar with GraphQL, there is some excellent documentation available at (LINK) so you can brush up on your skills! The query below will return the menu links.
+
+```js
+query SiteQuery {
+  site {
+    siteMetadata {
+      title
+      menuLinks {
+        name
+        link
+      }
+    }
+  }
+}
+``` 
+
+When executing this query within the GraphiQL editor you see output that looks similar to the following:
+
+```js
+{
+  "data": {
+    "site": {
+      "siteMetadata": {
+        "title": "Gatsby Default Starter",
+        "menuLinks": [
+          {
+            "name": "home",
+            "link": "/"
+          },
+          {
+            "name": "page2",
+            "link": "/page-2"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+Perfect! You now have a way of obtaining data from the `gatsby-config.js` file,  let's continue by pulling this data into the layout using the query you just formed.
+
+### Pulling data inside the layout component
+
+Inside your project locate the `src/components` folder and navigate to the `layout.js` file. Within this layout component you should notice a component named `StaticQuery`. 
+
+StaticQuery is a new component introduced in Gatsby V2 and allows you to run GraphQL queries within your components, not just pages. It allows developers to colocate data with their components.
+
+Currently the layout component should look like the following:
+
+```js
+  <StaticQuery
+    query={graphql`
+      query SiteTitleQuery {
+        site {
+          siteMetadata {
+            title
+            menuLinks {
+              name
+              link
+            }
+          }
+        }
+      }
+    `} ...left out for brevity/>
+```
+
+Let's extend the query within this component to include the menuLinks meta data, so it looks as so: (**Note:** *The GraphQL query must be located within the StaticQuery component, rather than referencing a variable from elsewhere due to how Gatsby parses our project*).
```

Comment:
This will be fixed by https://github.com/gatsbyjs/gatsby/pull/7801, so can be removed :)